features/step_definitions/logging.rb

```diff
@@ -1056,3 +1056,125 @@
+      when "unicode"
+        template_file = "#{file_dir}/container_json_unicode_log_template.json"
+      else
+        template_file = "#{file_dir}/container_json_log_template.json"
+      end
+
+    unless  project.pods(by:user).size > 0 
+        step %Q/I run the :new_app client command with:/,table(%{
+           | file | #{template_file} |
+         })
+    end
+end
+
+Given /^I have index pattern #{QUOTED}$/ do | pattern_name |
+    step %Q/I perform the :kibana_index_pattern_exist web action with:/,table(%{
+        | index_pattern_name | #{pattern_name} |
+    })
+    unless @result[:success]
+        step %Q/I run the :go_to_kibana_management_page web action/
+        step %Q/I perform the :create_index_pattern_in_kibana web action with:/, table(%{
+            | index_pattern_name | "#{pattern_name}" |
+         })
+         raise "Failed to create index pattern #{pattern_name}" unless @result[:success]
+    end
+end
+
+Given /^cluster-admin create the #{QUOTED} pattern for the #{WORD} user$/ do | pattern_name, who |
+    user(word_to_num(who))
+    user_name=user.name
+    user_token=user.cached_tokens.first
+    step %Q/I switch to cluster admin pseudo user/
+    step %Q/I use the "openshift-logging" project/
+    
+    rest_cmd="curl -s --connect-timeout 10  -XPOST \"http://localhost:5601/api/saved_objects/index-pattern/#{pattern_name}\" -H \"kbn-xsrf: true\" -H \"x-forwarded-user: #{user_name}\" -H \"securitytenant: __user__\" -H \"Authorization: Bearer #{user_token}\" -H \"Content-Type: application/json\" -d \'{\"attributes\": {\"title\": \"#{pattern_name}\", \"timeFieldName\": \"@timestamp\"}}\'"
+
+    step %Q/a pod becomes ready with labels:/, table(%{
+        | component= kibana|
+    })
+    @result = pod.exec("bash", "-c", rest_cmd, as: admin, container: 'kibana')
+    raise "Failed to create index pattern via rest api" unless @result[:success]
+end
+
+Given /^I can display the(?: "(.+)")? pod logs of the(?: "(.+)")? project under the(?: "(.+)")? pattern in kibana$/ do | pod_name, project_name, pattern_name |
+    # I switch to cluster admin pseudo user
+    pod_name ||= "*"
+    project_name ||= "*"
+    pattern_name ||= "app"
+    #lucene_query_string = "kubernetes.namespace_name: #{project_name} and kubernetes.pod_name: #{pod_name} and @timestamp:[now-15m TO now]"
+    lucene_query_string = "kubernetes.namespace_name: #{project_name} and kubernetes.pod_name: #{pod_name}"
+    step %Q/I perform the :select_index_pattern_in_kibana web action with:/,table(%{
+             | index_pattern_name | #{pattern_name} |
+    })
+    raise("Can not select to pattern #{ pattern_name}!") unless @result[:success]
+
+    step %Q/I perform the :search_doc_in_kibana web action with:/,table(%{
+             | search_string | #{lucene_query_string} |
+    })
+    raise("Can not refresh the search #{lucene_query_string}!") unless @result[:success]
+
+    retries = 3
+    while true
```

Comment:
@anpingli I think here the max trier is 4. And using `while true` is too dangerous.

Reply:
```
    retries = 3
    while true
        puts retries
        if retries > 0
           puts("Can not find documents using clause under pattern ! retry.....")
        else
           raise("Can not find documents using clause under pattern ! after 3 retry,  abort")
        end
        retries -= 1
    end
```
executed above block got below output:
```
3
Can not find documents using clause under pattern ! retry.....
2
Can not find documents using clause under pattern ! retry.....
1
Can not find documents using clause under pattern ! retry.....
0
test.rb:7:in `<main>': Can not find documents using clause under pattern ! after 3 retry,  abort (RuntimeError)
```

Reply:
Why do you think that is dangerous? the cucushift can't handle while wisely?

Reply:
No, why do you use `true` here? Using `true` makes me think it's an endless loop.

Reply:
Use ' while true' enable us to abort the loop raising message. ( raise("Can not find documents using clause #{lucene_query_string} under pattern  #{ pattern_name}! after 3 retry,  abort")
)

Reply:
I still think using endless loop is dangerous, if you want to raise error after 3 retries, I suggest to change the code to:
```
retries = 3
succeed=false
while retries > 0
  retries -= 1
  step %Q/I run the :check_log_count web action/
  if @result[:success]
    succeed=true
    break
  else
    log("Can not find documents using clause #{lucene_query_string} under pattern  #{ pattern_name}! retry.....")
  end
end
raise("Can not find documents using clause #{lucene_query_string} under pattern  #{ pattern_name}! after 3 retry,  abort") unless succeed
```